botorch/optim/optimize_alternating.py

```diff
@@ -0,0 +1,587 @@
+        options: Dict with optimization settings. Supports `maxiter_discrete`, `tol`
+            and `init_batch_limit`.
+
+    Returns:
+        A tuple of two tensors: a (num_restarts x d)-dim tensor of optimized points
+            and a (num_restarts)-dim tensor of their respective acquisition values.
+    """
+    with torch.no_grad():
+        current_acqval = acq_function(current_x.unsqueeze(0))
+    for _ in range(options.get("maxiter_discrete", MAX_ITER_DISCRETE)):
+        x_neighbors = get_nearest_neighbors(
+            current_x=current_x.detach(), bounds=bounds, discrete_dims=discrete_dims
+        )
+        with torch.no_grad():
+            acq_vals = torch.cat(
+                [
+                    acq_function(X_.unsqueeze(-2))
+                    for X_ in x_neighbors.split(
+                        options.get("init_batch_limit", MAX_BATCH_SIZE)
+                    )
+                ]
+            )
+        argmax = acq_vals.argmax()
+        improvement = acq_vals[argmax] - current_acqval
+        if improvement > 0:
+            current_acqval, current_x = acq_vals[argmax], x_neighbors[argmax]
+        if improvement <= options.get("tol", CONVERGENCE_TOL):
+            break
+    return current_x, current_acqval
+
+
+def continuous_step(
+    acq_function: AcquisitionFunction,
+    bounds: Tensor,
+    discrete_dims: Tensor,
+    current_x: Tensor,
+    options: Dict[str, Any],
+) -> Tuple[Tensor, Tensor]:
+    """Continuous search using L-BFGS-B through optimize_acqf.
+
+    Args:
+        acq_function: An AcquisitionFunction.
+        bounds: A `2 x d` tensor of lower and upper bounds for each column of `X`.
+        discrete_dims: A tensor of indices corresponding to binary and
+            integer parameters.
+        current_x: Starting point. A tensor of shape `d`.
+        options: Dict with optimization settings. Supports
+            `maxiter_continuous`, `tol` and `batch_limit`.
+
+    Returns:
+        A tuple of two tensors: a (1 x d)-dim tensor of optimized points
+            and a (1)-dim tensor of acquisition values.
+    """
+    if (current_x < bounds[0]).any() or (current_x > bounds[1]).any():
+        raise ValueError("continuous_step requires current_x to be within bounds.")
+    if len(discrete_dims) == len(current_x):  # nothing continuous to optimize
+        with torch.no_grad():
+            return current_x, acq_function(current_x.unsqueeze(0))
+
+    options = options or {}
+    return optimize_acqf(
+        acq_function=acq_function,
+        bounds=bounds,
+        q=1,
+        num_restarts=1,
+        batch_initial_conditions=current_x.unsqueeze(0),
+        fixed_features=dict(zip(discrete_dims.tolist(), current_x[discrete_dims])),
+        options={
+            "maxiter": options.get("maxiter_continuous", MAX_ITER_CONT),
+            "tol": options.get("tol", CONVERGENCE_TOL),
+            "batch_limit": options.get("batch_limit", MAX_BATCH_SIZE),
+        },
```

Comment:
```
options={
            "maxiter": options.get("maxiter_continuous", MAX_ITER_CONT),
            "tol": options.get("tol", CONVERGENCE_TOL),
            "batch_limit": options.get("batch_limit", MAX_BATCH_SIZE),
        },
```

These options gets passed as part of the `options` dictionary to scipy's optimize.minimize(...) in `botorch.optim.utils.timeout.py`:

```
return optimize.minimize(
            fun=fun,
            x0=x0,
            args=args,
            method=method,
            jac=jac,
            hess=hess,
            hessp=hessp,
            bounds=bounds,
            constraints=constraints,
            tol=tol,
            callback=wrapped_callback,
            options=options,
        )
```

In other words, `tol` here is `None`. To change the tolerance of the lbfgsb minimizer, "gtol" and/or "ftol" should be passed in:

```
options={
            "maxiter": options.get("maxiter_continuous", MAX_ITER_CONT),
            "gtol": options.get("tol", CONVERGENCE_TOL),
            "ftol": options.get("tol", CONVERGENCE_TOL),
            "batch_limit": options.get("batch_limit", MAX_BATCH_SIZE),
        },
```